scripts/apps/analytics/directives/ActivityReportPanel.js

```diff
@@ -0,0 +1,121 @@
+                        delete scope.activityReport.group_by;
+                        scope.activityReport.desk = desks.activeDeskId;
+                    }
+                }
+            });
+
+            /**
+             * Initialises the activity report object
+             */
+            scope.initActivityReport = function() {
+                scope.activityReport = {operation: 'publish', desk: desks.activeDeskId};
+                scope.group_by = {desk: false};
+            };
+
+            /**
+             * Returns true if the report editing tab was selected
+             */
+            scope.editReportSelected = function() {
+                return scope.panelTab === 'editingActivityReport';
+            };
+
+            /**
+             * Returns true if the saved reports tab was selected
+             */
+            scope.savedReportsSelected = function() {
+                return scope.panelTab !== 'editingActivityReport';
+            };
+
+            /**
+             * Changes the tab to the given one
+             * @param {String} tabName - valid values are 'editingActivityReport' and 'savedReports'
+             */
+            scope.changeTab = function(tabName) {
+                scope.panelTab = tabName;
+            };
+
+            /**
+             * Changes the inner tab to the given one
+             * @param {String} tabName - valid values are 'parameters' and 'grouping'
+             */
+            scope.display = function(tabName) {
+                scope.innerTab = tabName;
+            };
+
+            /**
+             * Generate the report
+             */
+            scope.generate = function() {
+                function onSuccess(activityReport) {
+                    $rootScope.$broadcast('view:activity_report', activityReport);
+                    notify.success(gettext('The activity report was genereated successfully'));
+                }
+
+                function onFail(error) {
+                    if (angular.isDefined(error.data._message)) {
+                        notify.error(error.data._message);
+                    } else {
+                        notify.error(gettext('Error. The activity could not be generated.'));
+                    }
+                }
+
+                var activityReportEdit = _.clone(scope.activityReport);
+                var toDelete = ['_id', '_etag', 'is_global', 'owner', 'name', 'description'];
+
+                activityReportEdit.operation_date = formatDate(scope.activityReport.operation_date);
+                toDelete.forEach((field) => {
+                    delete activityReportEdit[field];
+                });
+
+                api('activity_reports', session.identity).save({}, activityReportEdit)
+                    .then(onSuccess, onFail);
+            };
+
+            /**
+             * Format given date for generate
+             */
+            function formatDate(date) {
```

Comment:
Few places we are using `formatdate`. I might be a `filter` for formatting dates.

Moreover, all dates are stored as UTC and there is 10/11 hours difference for Sydney time. We would need timezone in `operation_date` otherwise we might not get the correct results.